sqlglot/dialects/snowflake.py

```diff
@@ -289,6 +296,16 @@ class Parser(parser.Parser):
             ),
         }
 
+        STATEMENT_PARSERS = {
+            **parser.Parser.STATEMENT_PARSERS,
+            TokenType.SHOW: lambda self: self._parse_show(),
+        }
+
+        SHOW_PARSERS = {
+            "PRIMARY KEYS": _show_parser("PRIMARY KEYS"),
+            "TERSE PRIMARY KEYS": _show_parser("PRIMARY KEYS"),
```

Comment:
The `TERSE` doesn't do anything here:

<img width="868" alt="Screenshot 2023-09-05 at 12 46 33 PM" src="https://github.com/tobymao/sqlglot/assets/46752250/2c4a0487-d83a-4c7d-bb88-4607cfbccc85">